test/topology_random_failures/cluster_events.py

```diff
@@ -0,0 +1,624 @@
+    await random_tables.drop_table(table=table)
+
+    yield
+
+    LOGGER.info("Check if the table was dropped by re-creating it")
+    await random_tables.add_table(ncolumns=5, name=table_name)
+
+
+async def add_index(manager: ManagerClient,
+                    random_tables: RandomTables,
+                    error_injection: str) -> AsyncIterator[None]:
+    yield
+
+    LOGGER.info("Add an index to a table")
+    index_name = await random_tables[0].add_index(column=random_tables[0].columns[-1])
+
+    yield
+
+    LOGGER.info("Check if the index was created by dropping it")
+    await random_tables[0].drop_index(name=index_name)
+
+
+async def drop_index(manager: ManagerClient,
+                     random_tables: RandomTables,
+                     error_injection: str) -> AsyncIterator[None]:
+    index_name = "test_random_failures_index_to_drop"
+
+    LOGGER.info("Add an index to a table")
+    await random_tables[0].add_index(column=random_tables[0].columns[-2], name=index_name)
+
+    yield
+
+    LOGGER.info("Drop the index from the table")
+    await random_tables[0].drop_index(name=index_name)
+
+    yield
+
+    LOGGER.info("Check if the index was dropped by re-creating it")
+    await random_tables[0].add_index(column=random_tables[0].columns[-2], name=index_name)
+
+
+async def add_new_keyspace(manager: ManagerClient,
+                           random_tables: RandomTables,
+                           error_injection: str) -> AsyncIterator[None]:
+    ks_name = "test_random_failures_new_ks"
+
+    yield
+
+    LOGGER.info("Add a new keyspace to the schema")
+    await manager.cql.run_async(
+        f"CREATE KEYSPACE {ks_name}"
+        f" WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 3 }}"
+    )
+
+    yield
+
+    LOGGER.info("Check if the keyspace exists by dropping it")
+    await manager.cql.run_async(f"DROP KEYSPACE {ks_name}")
+
+
+async def drop_keyspace(manager: ManagerClient,
+                        random_tables: RandomTables,
+                        error_injection: str) -> AsyncIterator[None]:
+    ks_name = "test_random_failures_ks_to_drop"
+
+    LOGGER.info("Add a keyspace to drop")
+    await manager.cql.run_async(
+        f"CREATE KEYSPACE {ks_name}"
+        f" WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 3 }}"
+    )
+
+    yield
+
+    LOGGER.info("Drop the keyspace from the schema")
+    await manager.cql.run_async(f"DROP KEYSPACE {ks_name}")
+
+    yield
+
+    LOGGER.info("Check if the keyspace was dropped by re-creating it")
+    await manager.cql.run_async(
+        f"CREATE KEYSPACE {ks_name}"
+        f" WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 3 }}"
+    )
+
+
+@deselect_for(
+    # TODO: remove this skip when #16317 will be resolved.
+    reason="Cannot create CDC log for tables, because keyspace uses tablets. See issue #16317",
```

Comment:
So the test always runs in tablets mode?

Reply:
We should test CDC in non-tablets mode too. (Especially because CDC in non-tablets mode is already working)

Reply:
Enabling tablets could be a randomly chosen parameter BTW

Reply:
We plan to extend number of tests. It's a good candidate for another dimension.